engine/src/main/java/com/datatorrent/stram/engine/Node.java

```diff
@@ -485,6 +485,10 @@ protected void deactivateSinks()
 
   void checkpoint(long windowId)
   {
+    if (operator instanceof Operator.CheckpointNotificationListener) {
```

Comment:
I agree with @PramodSSImmaneni 
By definition a Stateless operator is an operator which doesn't have a state to persist which is irrespective of whether the operator developer wants the out-of-box state persistence by platform or use any other custom way of persistence. But if the operator developer marks it @Stateless that indicates that there isn't any state. 
Being Stateless doesn't mean that operator will handle persisting state in a custom way. For that we have a pluggable storage agent API.